components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/artifact/SAMLSSOArtifactResolutionService.java

```diff
@@ -0,0 +1,257 @@
+import org.opensaml.saml2.core.ArtifactResponse;
+import org.opensaml.saml2.core.Issuer;
+import org.opensaml.saml2.core.StatusCode;
+import org.opensaml.ws.soap.soap11.Envelope;
+import org.opensaml.xml.XMLObjectBuilderFactory;
+import org.wso2.carbon.identity.sso.agent.bean.SSOAgentConfig;
+import org.wso2.carbon.identity.sso.agent.exception.ArtifactResolutionException;
+import org.wso2.carbon.identity.sso.agent.exception.SSOAgentException;
+import org.wso2.carbon.identity.sso.agent.security.X509CredentialImpl;
+import org.wso2.carbon.identity.sso.agent.util.SSOAgentUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.UUID;
+import javax.xml.soap.MessageFactory;
+import javax.xml.soap.MimeHeaders;
+import javax.xml.soap.SOAPBody;
+import javax.xml.soap.SOAPBodyElement;
+import javax.xml.soap.SOAPException;
+import javax.xml.soap.SOAPMessage;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+/**
+ * This class is used for handling SAML2 Artifact Binding.
+ */
+public class SAMLSSOArtifactResolutionService {
+
+    private static Log log = LogFactory.getLog(SAMLSSOArtifactResolutionService.class);
+
+    private SSOAgentConfig ssoAgentConfig;
+
+    public SAMLSSOArtifactResolutionService(SSOAgentConfig ssoAgentConfig) {
+
+        this.ssoAgentConfig = ssoAgentConfig;
+    }
+
+    /**
+     * Create a SAML artifactResolveObject based on given samlArt parameter and wrapped into a soapRequest
+     * Send the soapRequest to the Artifact Resolve Endpoint
+     *
+     * @param samlArtReceived SAML Artifact reference needed to get the actual data
+     * @return ArtifactResponse
+     */
+    public ArtifactResponse getSAMLArtifactResponse(String samlArtReceived) throws ArtifactResolutionException {
+
+        validateArtifactResolveConfig();
+        ArtifactResolve artifactResolve = generateArtifactResolveReq(samlArtReceived);
+        return sendArtifactResolveRequest(artifactResolve);
+    }
+
+    /**
+     * Create SAML ArtifactResolve Object and sign
+     *
+     * @param samlArtReceived object reference to actual data
+     * @return SAML ArtifactResolve Object
+     */
+    public ArtifactResolve generateArtifactResolveReq(String samlArtReceived) throws ArtifactResolutionException {
+
+        ArtifactResolve artifactResolve = createArtifactResolveObject(samlArtReceived);
+        if (ssoAgentConfig.getSAML2().isEnableArtifactResolveSigning()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Signing artifact resolve request for the received SAML artifact: " + samlArtReceived);
+            }
+            artifactResolve = signArtifactResolveReq(artifactResolve);
+        }
+        return artifactResolve;
```

Comment:
Would be better to include debug logs to print samlArtReceived and created ArtifactResolve request.

Reply:
ArtifactResolve is printed in [here](https://github.com/wso2-extensions/identity-agent-sso/pull/46/files#diff-e58b1bc8dd5cd23b7eb0aa3b3e5d0a68R129) along with SOAP envelope. Should we print it here as well?